pandas/core/indexes/base.py

```diff
@@ -2379,6 +2379,10 @@ def unique(self, level=None):
         """
         if level is not None:
             self._validate_index_level(level)
+
+        if self.is_unique:
+            return self
```

Comment:
this needs to return self._shallow_copy()

Reply:
Thx.